src/test/java/com/project/playlist/integration_testing/PlaylistControllerIntegrationTest.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class PlaylistControllerIntegrationTest extends MySqlIntegrationTest {
+    @Autowired
+    PlaylistRepository playlistRepository;
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    VideoRepository videoRepository;
+    @Autowired
+    PlaylistVideoRepository playlistVideoRepository;
+    @Autowired
+    ObjectMapper objectMapper;
+    @Autowired
+    private MockMvc mvc;
+
+    private User user;
+    private Video video1, video2;
+    private Playlist playlist1, playlist2;
+    private PlaylistVideo pv1, pv2;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        user = userRepository.save(User.builder().username("username").email("email").password("pass").build());
+        video1 = videoRepository.save(Video.builder().url("url1").name("videoName1").description("videoDescription1").user(user).build());
+        video2 = videoRepository.save(Video.builder().url("url2").name("videoName2").description("videoDescription2").user(user).build());
+        playlist1 = playlistRepository.save(Playlist.builder().user(user).name("playlistName1").build());
+        pv1 = playlistVideoRepository.save(PlaylistVideo.builder().playlist(playlist1).video(video1).orderNo(1).build());
+        pv2 = playlistVideoRepository.save(PlaylistVideo.builder().playlist(playlist1).video(video2).orderNo(2).build());
+        playlist1.setPlaylistVideos(new ArrayList<>(List.of(pv1, pv2)));
+        playlist1 = playlistRepository.save(playlist1);
+        playlist2 = playlistRepository.save(Playlist.builder().name("playlistName2").user(user).build());
+    }
+
+    @Test
+    public void givenExistingPlaylistId_whenGetPlaylistById_thenReturnPlaylistWithVideos() throws Exception {
+        //arrange
+        String playlistId = String.valueOf(playlist1.getId());
+        //act
+        MvcResult result = mvc.perform(get("/api/playlists/" + playlistId).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        String jsonContent = result.getResponse().getContentAsString();
+        PlaylistDTO playlistDto = objectMapper.readValue(jsonContent, PlaylistDTO.class);
+        Playlist playlist = playlistRepository.findById(playlistDto.getId()).get();
+        //assert
+        assertAll(
+                () -> assertEquals("playlistName1", playlist.getName()),
+                () -> assertEquals("playlistName1", playlistDto.getName()),
```

Comment:
You don't need to check playListDto, you need to check what you retrieved from database if those values are correct